API/testrunner/devices/device_base.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2018-present Samsung Electronics Co., Ltd. and other contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+
+from API.common import console, utils, paths
+from threading import Thread
+
+class RemoteDevice(object):
+    '''
+    Base class of all the device classes.
+    '''
+    def __init__(self, env):
+        self.app = env['info']['app']
+        self.user = env['info']['username']
+        self.ip = env['info']['ip']
+        self.port = env['info']['port']
+        self.device = env['info']['device']
+        self.env = env
+
+        self.check_args()
+
+    def check_args(self):
+        '''
+        Check that all the arguments are established.
+        '''
+        if self.device in ['rpi2', 'artik530']:
+            if not self.workdir:
+                console.fail('Please use --remote-workdir for the device.')
+            if not self.ip:
+                console.fail('Please define the IP address of the device.')
+            if not self.user:
+                console.fail('Please define the username of the device.')
+
+            if self.workdir is '/':
+                console.fail('Please do not use the root folder as test folder.')
+
+        else:
```

Comment:
Maybe it would be better to use the `elif self.device in ['artik053', 'stm32f4dis']` condition here and add an else case with the following `console.fail('The selected device is not supported')`. So we will not forget to check the arguments here in case of a new device.
On the other hand,  the possible argument choices in `driver.py` does not let us to select other devices, so this will be an extra check.
@rtakacs : What is your opinion?

Reply:
I agree with @robertsipka.

Reply:
We could use ``assert``, but the output format of ``console.fail()`` is better in my opinion, because ``assert`` outputs a backtrace by default which is not really needed here, and maybe even distracting. Maybe we could use something like this:
```
elif self.device in ['artik053', 'stm32f4dis']:
    try:
        assert self.env['info']['device_id']

     except AssertionError:
         console.fail('Please use the --device-id to select the device')
```
But I don't know if this is better than using a simple if statement.